expression/builtin_time_test.go

```diff
@@ -370,20 +370,21 @@ func (s *testEvaluatorSuite) TestDayOfMonth(c *C) {
 	defer testleak.AfterTest(c)()
 	sc := s.ctx.GetSessionVars().StmtCtx
 	sc.IgnoreZeroInDate = true
```

Comment:
IgnoreZeroInDate is true, what's the problem with the test and why you change it?

Reply:
@tiancaiamao the changes are to test different combination of sql_mode `NO_ZERO_DATE` and `STRICT_TRANS_TABLES`. `IgnoreZeroInDate` actually don't represent these sql_modes